# Hypothesis Testing and Chi-Square Test(Assignment)

# 1. CHI-SQUARE TEST

## Objective

To use the Chi-Square test for independence to determine if there's a significant association between the type of smart home device purchased (Smart Thermostats vs. Smart Lights) and the customer satisfaction level.

## Task 1: State the Hypotheses

**Null Hypothesis (H₀):**

There is no association between the type of device and customer satisfaction, i.e. customer satisfaction is independent of the type of device purchased.
                     


**Alternative Hypothesis (H₁):**

There is an association between the type of device and customer satisfaction, i.e. satisfaction level depends on the type of device purchased.

## Task 2: Compute the Chi-Square Statistic

In [2]:
import numpy as np
import scipy.stats as stats

In [3]:
# Observed frequency
observed = np.array([
    [50, 70],
    [80, 100],
    [60, 90],
    [30, 50],
    [20, 50]
])

In [4]:
chi2, p, dof, expected = stats.chi2_contingency(observed)

In [5]:
# Results
print(f'Chi-Square Statistic: {chi2:.4f}')
print(f'P-Value: {p:.4f}')
print(f'Degrees of Freedom: {dof}')
print("Expected Frequencies Table:")
print(expected)

Chi-Square Statistic: 5.6382
P-Value: 0.2278
Degrees of Freedom: 4
Expected Frequencies Table:
[[ 48.  72.]
 [ 72. 108.]
 [ 60.  90.]
 [ 32.  48.]
 [ 28.  42.]]


## Task 3: Determine the Critical Value

In [6]:
alpha = 0.05

In [9]:
# Degrees of freedom (df) = (rows - 1) * (columns - 1)
dof = (5 - 1) * (2 - 1)
dof

4

In [8]:
critical_value = stats.chi2.ppf(1 - alpha, dof)
print(f"Critical Value: {critical_value:.4f}")

Critical Value: 9.4877


## Task 4: Make a Decision

The Chi-Square Statistic is 5.6382 and Critical Value is 9.4877

Since 5.64 < 9.488, we fail to reject null hypothesis.

This means, there is no significant association between the type of smart home device and customer satisfaction at the 5% significance level.

# 2. HYPOTHESIS TESTING

## Objective

To investigate the restaurant owners' claim about the increase in weekly operating costs using hypothesis testing.

## Task 1: State the Hypotheses 

Null Hypothesis (H₀): The actual mean weekly operating cost follows the given cost model.

**H₀: μ = 4000**

Alternative Hypothesis (H₁): The actual mean weekly operating cost is higher than the predicted cost.

**H₁: μ < 4000**

## Task 2: Calculate the Test Statistic

ˉxˉ = sample mean weekly cost (Rs. 3,050) # Observed_mean

**Expected Mean:**
μ = theoretical mean weekly cost according to the cost model (W = $1,000 + $5X for X = 600 units) = (1,000 + 5 × 600) = 4000 

σ = 5*25 units

n = sample size (25 restaurants)

In [12]:
observed_mean = 3050
expected_mean = 4000
sigma = 5 * 25  # 5 * 25 units
sample_size = 25

### Calculate Standard Error

In [13]:
# alpha level of 5% (α = 0.05) 

se = sigma / (sample_size ** 0.5)

### Calculate the t-statistic

In [14]:
t_statistic = (observed_mean - expected_mean) / se
print(f"T-Statistic: {t_statistic:.2f}")

T-Statistic: -38.00


## Task 3: Determine the Critical Value

In [15]:
from scipy.stats import norm

# Significance level (alpha)
alpha = 0.05

# Critical value for a one-tailed test (right tail)
critical_value = norm.ppf(1 - alpha)

print(f"Critical Value (Z): {critical_value:.4f}")

Critical Value (Z): 1.6449


## Task 4: Make a decision

The test statistic (t) is -38, and the critical value (Z) is 1.645.

Since -38 < 1.645, we reject null hypothesis.

This means, the actual mean weekly operating cost does not follow the given cost model.

## Task 5: Conclusion

Based on Task 4, there is strong evidence to support the restaurant owners' claim that the weekly operating costs are higher than the model suggests